internal/smartyresponseprocessing/smartyresponseprocessing.go

```diff
@@ -0,0 +1,81 @@
+package smartyresponseprocessing
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type SmartyResponse struct {
+	Zipcodes []struct {
+		Zipcode           string  `json:"zipcode"`
+		ZipcodeType       string  `json:"zipcode_type"`
+		DefaultCity       string  `json:"default_city"`
+		CountyFIPS        string  `json:"county_fips"`
+		CountyName        string  `json:"county_name"`
+		StateAbbreviation string  `json:"state_abbreviation"`
+		State             string  `json:"state"`
+		Latitude          float64 `json:"latitude"`
+		Longitude         float64 `json:"longitude"`
+		Precision         string  `json:"precision"`
+		AlternateCounties []struct {
+			CountyFIPS        string `json:"county_fips"`
+			CountyName        string `json:"county_name"`
+			StateAbbreviation string `json:"state_abbreviation"`
+			State             string `json:"state"`
+		} `json:"alternate_counties"`
+	} `json:"zipcodes"`
+}
+
+type ZipcodeResult struct {
+	Zipcode         string
+	USPSFips        []string
+	SmartyFips      []string
+	Inconsistencies int
+}
+
+func ProcessSmartyResponse(responseBody []byte, zipToCounties map[string][]string, yield func(ZipcodeResult)) error {
+	var smartyResponses []SmartyResponse
+	err := json.Unmarshal(responseBody, &smartyResponses)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	for _, response := range smartyResponses {
+		for _, zipcode := range response.Zipcodes {
+			result := ZipcodeResult{
+				Zipcode:  zipcode.Zipcode,
+				USPSFips: zipToCounties[zipcode.Zipcode],
+			}
+
+			// Extract the last three digits of the Smarty Fips code, which represent the county.
+			smartyFIPS := []string{zipcode.CountyFIPS[len(zipcode.CountyFIPS)-3:]}
+			for _, altCounty := range zipcode.AlternateCounties {
+				smartyFIPS = append(smartyFIPS, altCounty.CountyFIPS[len(altCounty.CountyFIPS)-3:])
+			}
+			result.SmartyFips = smartyFIPS
+
+			inconsistencies := calculateInconsistencies(result.USPSFips, result.SmartyFips)
+			result.Inconsistencies = inconsistencies
+
+			yield(result)
+		}
+	}
+
+	return nil
+}
+
+func calculateInconsistencies(uspsFIPS, smartyFIPS []string) int {
```

Comment:
This will count the number of counties that are in Smarty but not in the USPS. We want to go the other way too — counties in USPS but not in Smarty. One way to find all diffs would be to sort and compare the slices. The functions in the [slices](https://pkg.go.dev/slices) package (`Sort`, `Equal`, etc.) could help.